### ISSUE TITLE
default function in schema without context

### ISSUE THREAD

**Issue by kof** [Automattic/mongoose#1581]
hi,

default function in the schema has currently no way to access the document.

```
var m = require('mongoose');

var schema = new m.Schema({
    a: {
        type: Number,
        default: function() {
            console.log(this);
        }
    }
});

var Model = m.model('mymodel', schema);

new Model();
```

**Comment by yunnysunny** [comment]
+1

**Comment by josemontesp** [comment]
+1

**Comment by siboulet** [comment]
Is there any plans on adding this? Not sure if thats expected or actually a bug, noticed this issue was opened back in 2013. Confirmed was an issue with 4.x and still an issue on 5.x.

```js
var assert = require('assert');
var mongoose = require('mongoose');

var schema = new mongoose.Schema({
   title: String,
   arnold: Boolean,
});

schema.path('arnold').default(function() {
  return this.title && this.title.match(/arnold/i);
});

var BlogPost = mongoose.model('BlogPost', schema);

var post = new BlogPost({title: '5 Best Arnold Schwarzenegger Movies'});

assert.equal(post.arnold, true);

```

**Comment by lineus** [comment]
This was resolved in #6119. 

### 1581.js
```js
#!/usr/bin/env node
'use strict';

const assert = require('assert');
const mongoose = require('mongoose');
const { Schema } = mongoose;

const schema = new Schema({
  first: String,
  last: String,
  revName: String,
  fullName: {
    type: String,
    default: function() {
      if (allStrings(this.first, this.last)) {
        return `${this.first} ${this.last}`;
      }
    }
  }
});

schema.path('revName').default(function() {
  if (allStrings(this.first, this.last)) {
    return `${this.last}, ${this.first}`;
  }
});

const Test = mongoose.model('test', schema);

const test = new Test({ first: 'test', last: 'ing' });

assert.strictEqual(test.fullName, 'test ing');
assert.strictEqual(test.revName, 'ing, test');
console.log('All Assertions Pass.');

function allStrings() {
  let ret = true;
  [...arguments].map(arg => {
    if (typeof arg !== 'string') {
      ret = false;
    }
  });
  return ret;
}
```

### Output
```
issues: ./1581.js 
All Assertions Pass.
issues: 
```
